### PR TITLE
Improve Kruskal and use in-built disjoint set data structure

### PR DESCRIPTION
**Pull request by SohamTamba**
Improvements due to sorting:

For a complete graph with 1,000 vertices, the run-time improved from 193 ms to 28 ms.
For a path graph with 10,000 vertices, the run-time improved from 3.88 ms to 1.15 ms.

I removed the disjoint set data structure I had created and replaced it with the Inbuilt Disjoint Set. because the improvement in performance was little (28ms to to 26ms)

### CODE REVIEW

**Comment by sbromberger** [comment]
@SohamTamba - thanks for this! 
@somil55 - for your review / decision, please. If you say OK, then I'll be happy to merge.

**Comment by somil55** [comment]
@sbromberger Will read through this by tonight.

File: src/spanningtrees/kruskal.jl [hunk 1]
```diff
@@ -1,70 +1,26 @@
...
-end
-
-union_set!(set_id::Vector{U}, p::U, q::U) where {U<:Integer} = (set_id[find_set!(set_id, p)] = find_set!(set_id, q))
-
-"""
-    kruskal_mst(g, distmx=weights(g))
-
-Return a vector of edges representing the minimum spanning tree of a connected, undirected graph `g` with optional
-distance matrix `distmx` using [Kruskal's algorithm](https://en.wikipedia.org/wiki/Kruskal%27s_algorithm).
-"""
```
Comment (somil55) [inline src/spanningtrees/kruskal.jl]: Add doc string

File: src/spanningtrees/kruskal.jl [hunk 2]
```diff
@@ -1,70 +1,26 @@
...
-        if find_set!(connected_vs, v) != find_set!(connected_vs, w)
-            union_set!(connected_vs, v, w)
-            push!(mst, heap_entry.edge)
+    for e in edge_list[sortperm(weights)]
+        if !DataStructures.in_same_set(connected_vs, e.src, e.dst)
+            DataStructures.union!(connected_vs, e.src, e.dst)
+            push!(mst, e)
             if length(mst) >= nv(g) - 1
                 break
             end
```
Comment (somil55) [inline src/spanningtrees/kruskal.jl]: Put a shorthand notation: `(length(mst)>= nv(g)-1 ) && break `

**Comment by somil55** [comment]
@sbromberger Please merge this!

**Pull request merged**